packages/firestore/src/platform/browser_lite/fetch_connection.ts

```diff
@@ -69,9 +69,23 @@ export class FetchConnection extends RestConnection {
     }
 
     if (!response.ok) {
+      let errorMessage;
+      try {
+        // Try to get any error message text from server response.
+        //response.json returns an error object inside an array.
+        const [errorResponse] = (await response.json()) as [
```

Comment:
Reduce the try/catch block to _only_ contain the call to `await response.json()`.

Reply:
As I am not 100% confident that the response.json() will always return [ {error?: { message?: string };}];, would it be better to keep it in the try/catch block?  If it succeed, display errorMessage. If it fail, then display statusText as minimum

Reply:
It would be better to guard it with runtime checks than rely on exceptions because it's more explicit about what you are doing, and, therefore, more readable.

Maybe do something like this:
```js
let errorResponse;
try {
  errorResponse = await response.json();
} catch (_ignored) {
}

let errorMessage;
if (errorResponse) {
  errorMessage = Array.from(errorResponse).shift()?.error?.message
}
```